packages/typegpu/src/core/function/tgpuComputeFn.ts

```diff
@@ -36,17 +38,31 @@ export interface TgpuComputeFn extends TgpuResolvable, TgpuNamable {
   $__ast(argNames: string[], body: Block): this;
 }
 
+export interface ComputeFnOptions {
+  workgroupSize: number[];
+}
+
 /**
  * Creates a shell of a typed entry function for the compute shader stage. Any function
  * that implements this shell can perform general-purpose computation.
  *
  * @param workgroupSize
  *   Size of blocks that the thread grid will be divided into (up to 3 dimensions).
  */
-export function computeFn(workgroupSize: number[]): TgpuComputeFnShell {
+export function computeFn<Args extends AnyBuiltin[] | []>(
+  argTypes: Args,
+  options: ComputeFnOptions,
+): TgpuComputeFnShell {
```

Comment:
I don't think we need to extend `[]`. Do we need this template? Is `argTypes` only here to create a type hint?

Reply:
The `| []` is forcing the template to be a tuple instead of an array of a union. Though yeah, I think the template can be removed altogether as you suggest 💜.